Projects/App/Sources/Result/ResultViewController.swift

```diff
@@ -0,0 +1,142 @@
+        view.addSubview(resultView)
+        
+        resultView.snp.makeConstraints {
+            $0.top.equalTo(navigationBar.snp.bottom)
+            $0.leading.trailing.equalToSuperview()
+            $0.bottom.equalTo(view.safeAreaLayoutGuide)
+        }
+    }
+    
+    override func configure() {
+        super.configure()
+        
+        view.backgroundColor = .designSystem(.primaryYellow)
+        
+        configureNavigationBar()
+        
+        navigationBar.leftButtonTapEvent.subscribe(onNext: { [weak self] in
+            self?.dismiss(animated: true)
+        }).disposed(by: disposeBag)
+        
+        resultView.delegate = self
+        resultView.type = viewModel.type
+        resultView.iconDelegate = self
+    }
+    
+    private func configureNavigationBar() {
+        view.addSubview(navigationBar)
+        
+        navigationBar.snp.makeConstraints {
+            $0.top.equalTo(view.safeAreaLayoutGuide)
+            $0.leading.trailing.equalToSuperview()
+        }
+    }
+}
+
+extension ResultViewController: ResultViewButtonDelegate {
+    func homeButtonTapped() {
+        dismiss(animated: true)
+    }
+    
+    func saveButtonTapped() {
+        // TODO: 서버에서 받아온 이미지로 저장해줘야함
+        guard let image = UIImage(named: "img_gosim3") else { return }
+        UIImageWriteToSavedPhotosAlbum(image, self, #selector(gifticonImageSaved), nil)
+    }
+    
+    @objc private func gifticonImageSaved(image: UIImage, error: Error?, context: UnsafeRawPointer) {
+        dimView.backgroundColor = UIColor.black.withAlphaComponent(0.25)
+        dimView.addSubview(toastView)
+        wrapperView.addSubview(dimView)
+        
+        let tapGesture = UITapGestureRecognizer(target: self, action: #selector(dismissPopupView(_:)))
+        wrapperView.addGestureRecognizer(tapGesture)
+        wrapperView.isUserInteractionEnabled = true
+        
+        view.addSubview(wrapperView)
+        wrapperView.snp.makeConstraints {
+            $0.edges.equalToSuperview()
+        }
+        
+        dimView.snp.makeConstraints {
+            $0.edges.equalToSuperview()
+        }
+        
+        toastView.snp.makeConstraints {
+            $0.width.equalTo(289)
+            $0.height.equalTo(295)
+            $0.centerX.centerY.equalToSuperview()
+        }
+
+        toastView.alpha = 0
+        
+        DispatchQueue.main.asyncAfter(deadline: DispatchTime.now()) {
+            UIView.transition(with: self.view, duration: 0.3, options: [.curveEaseInOut], animations: {
+                self.toastView.alpha = 1
+            }, completion: nil)
+        }
+        
+        DispatchQueue.main.asyncAfter(deadline: DispatchTime.now() + 4) {
+            UIView.transition(with: self.view, duration: 0.3, options: [.curveEaseInOut], animations: {
+                self.toastView.alpha = 0
+                self.dimView.removeFromSuperview()
+                self.wrapperView.removeFromSuperview()
+            }, completion: nil)
```

Comment:
animation 종료 이후에 호출되는 내용이라면 아래처럼 작성하면 이해하기 더 편합니다. 😃 
```suggestion
                self.toastView.alpha = 0
            }, completion: {
                self.dimView.removeFromSuperview()
                self.wrapperView.removeFromSuperview()
            })
```